src/shogun/machine/GaussianProcessMachine.cpp

```diff
@@ -79,12 +80,14 @@ SGVector<float64_t> CGaussianProcessMachine::get_posterior_means(CFeatures* data
 	CFeatures* feat;
 
 	// use inducing features for FITC inference method
-	if (m_method->get_inference_type()==INF_FITC)
+	if (m_method->get_inference_type()==INF_FITC_REGRESSION ||
+		m_method->get_inference_type()==INF_FITC_LAPLACIAN_SINGLE)
 	{
-		CFITCInferenceMethod* fitc_method=
-			CFITCInferenceMethod::obtain_from_generic(m_method);
+		CSingleFITCLaplacianBase* fitc_method=
+			dynamic_cast<CSingleFITCLaplacianBase *>(m_method);
+		REQUIRE(fitc_method, "Inference method %s must support FITC inference\n",
```

Comment:
Slight change:
"Inference method %s does not support FITC inference\n"

More easily understandable for user